demos/common/cpp/models/src/hpe_model_associative_embedding.cpp

```diff
@@ -0,0 +1,207 @@
+    cv::Mat resizedImage;
+    float scale = std::min(inputLayerSize.height / static_cast<float>(image.rows),
+                           inputLayerSize.width / static_cast<float>(image.cols));
+    cv::resize(image, resizedImage, cv::Size(), scale, scale, cv::INTER_CUBIC);
+    int h = resizedImage.rows;
+    int w = resizedImage.cols;
+    cv::Mat paddedImage;
+    int bottom = inputLayerSize.height - resizedImage.rows;
+    int right = inputLayerSize.width - resizedImage.cols;
+    cv::copyMakeBorder(resizedImage, paddedImage, 0, bottom, 0, right,
+                       cv::BORDER_CONSTANT, meanPixel);
+    Blob::Ptr frameBlob = request->GetBlob(inputsNames[0]);
+    matU8ToBlob<uint8_t>(paddedImage, frameBlob);
+    return std::shared_ptr<InternalModelData>(new InternalScaleData(image.cols / static_cast<float>(w),
+                                                                    image.rows / static_cast<float>(h)));
+}
+
+std::unique_ptr<ResultBase> HpeAssociativeEmbedding::postprocess(InferenceResult& infResult) {
+    HumanPoseResult* result = new HumanPoseResult;
+    *static_cast<ResultBase*>(result) = static_cast<ResultBase&>(infResult);
+
+    auto aembdsMapped = infResult.outputsData[outputsNames[0]];
+    auto heatMapsMapped = infResult.outputsData[outputsNames[1]];
+    auto nmsHeatMapsMapped = infResult.outputsData[outputsNames[2]];
+
+    const SizeVector& aembdsDims = aembdsMapped->getTensorDesc().getDims();
+    const SizeVector& heatMapsDims = heatMapsMapped->getTensorDesc().getDims();
+    const SizeVector& nmsHeatMapsDims = nmsHeatMapsMapped->getTensorDesc().getDims();
+
+    float* aembds = aembdsMapped->rmap().as<float*>();
+    float* heats = heatMapsMapped->rmap().as<float*>();
+    float* nmsHeats = nmsHeatMapsMapped->rmap().as<float*>();
+
+    std::vector<cv::Mat> heatMaps(heatMapsDims[1]);
+    convertTo3D(heatMaps, heats, heatMapsDims);
+
+    std::vector<cv::Mat> aembdsMaps(aembdsDims[1]);
+    convertTo3D(aembdsMaps, aembds, aembdsDims);
+
+    std::vector<cv::Mat> nmsHeatMaps(nmsHeatMapsDims[1]);
+    convertTo3D(nmsHeatMaps, nmsHeats, nmsHeatMapsDims);
+
+    std::vector<HumanPose> poses = extractPoses(heatMaps, aembdsMaps, nmsHeatMaps);
+
+    // Rescale poses to the original image
+    const auto& scale = infResult.internalModelData->asRef<InternalScaleData>();
+    float outputScale = inputLayerSize.width / static_cast<float>(heatMapsDims[3]);
+    float scaleX = scale.x * outputScale;
+    float scaleY = scale.y * outputScale;
+
+    for (auto& pose : poses) {
+        for (auto& keypoint : pose.keypoints) {
+            if (keypoint != cv::Point2f(-1, -1)) {
+                keypoint.x *= scaleX;
+                keypoint.y *= scaleY;
+                std::swap(keypoint.x, keypoint.y);
+            }
+        }
+        result->poses.push_back(pose);
+    }
+
+    return std::unique_ptr<ResultBase>(result);
+}
+
+void HpeAssociativeEmbedding::convertTo3D(std::vector<cv::Mat>& flattenData, float* data,
```

Comment:
Name it `split()` to follow OpenCV's naming. The function can construct and return `flattenData` on its own instead of filling preallocated one.